src/unix/signal.c

```diff
@@ -53,8 +53,13 @@ RB_GENERATE_STATIC(uv__signal_tree_s,
                    uv_signal_s, tree_entry,
                    uv__signal_compare)
 
+static void uv__signal_global_reinit(void);
 
 static void uv__signal_global_init(void) {
+  if (!uv__signal_lock_pipefd[0])
+    if (pthread_atfork(NULL, NULL, &uv__signal_global_reinit))
+      abort();
```

Comment:
`pthread_atfork` can register before and after handlers, one for each child. This only registers one for the child. That state is both persistent and cumulative, so if we keep doing it the handler functions will be called multiple times. Thus we only want to do it once.